lib/JsonpHotUpdateChunkTemplatePlugin.js

```diff
@@ -2,24 +2,27 @@
 	MIT License http://www.opensource.org/licenses/mit-license.php
 	Author Tobias Koppers @sokra
 */
-var ConcatSource = require("webpack-sources").ConcatSource;
+"use strict";
 
-function JsonpHotUpdateChunkTemplatePlugin() {}
-module.exports = JsonpHotUpdateChunkTemplatePlugin;
+const ConcatSource = require("webpack-sources").ConcatSource;
+
+class JsonpHotUpdateChunkTemplatePlugin {
+	apply(hotUpdateChunkTemplate) {
+		hotUpdateChunkTemplate.plugin("render", (modulesSource, modules, removedModules, hash, id) => {
+			const jsonpFunction = this.outputOptions.hotUpdateFunction;
+			const source = new ConcatSource();
+			source.add(`${jsonpFunction}(${JSON.stringify(id)},`);
+			source.add(modulesSource);
+			source.add(")");
+			return source;
+		});
+		hotUpdateChunkTemplate.plugin("hash", (hash) => {
+			hash.update("JsonpHotUpdateChunkTemplatePlugin");
+			hash.update("3");
+			hash.update(`${this.outputOptions.hotUpdateFunction}`);
```

Comment:
`this.outputOptions` is lost because of the arrow function. You can revert to normal, or do something like:

```
module.exports = class JsonpHotUpdateChunkTemplatePlugin {
	apply(hotUpdateChunkTemplate) {
		let outputOptions = this.outputOptions;
		hotUpdateChunkTemplate.plugin("render", (modulesSource, modules, removedModules, hash, id) => {
			const jsonpFunction = outputOptions.hotUpdateFunction;
			const source = new ConcatSource();
			source.add(`${jsonpFunction}(${JSON.stringify(id)},${modulesSource})`);
			return source;
		});
		hotUpdateChunkTemplate.plugin("hash", (hash) => {
			hash.update("JsonpHotUpdateChunkTemplatePlugin");
			hash.update("3");
			hash.update(`${outputOptions.hotUpdateFunction}`);
			hash.update(`${outputOptions.library}`);
		});
	}
}
```